### PR TITLE
Update CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by avie66**
Add models contribution guide

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,124 +1,424 @@
...
+## 🧪 Testing Your Contributions
+
+### Unit Tests
+
+```bash
+cd refact-agent/engine/
+cargo test
+```
+
+### Integration Tests
```
Comment (humbertoyusta) [inline CONTRIBUTING.md:270]: Integration tests are quite outdated, skip this part

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,124 +1,424 @@
...
+
+1. **API Keys**: Use environment variables for sensitive data
+2. **Endpoints**: Ensure URLs are correct and follow OpenAI compatibility
+3. **Error Handling**: Test edge cases and error conditions
+4. **Rate Limiting**: Consider provider-specific limitations
 
+### Code Quality
 
-#### Testing
+1. **Follow Rust conventions**: Use `cargo fmt` and `cargo clippy`
```
Comment (humbertoyusta) [inline CONTRIBUTING.md:313]: Remove this part, our project won't work with cargo clippy or cargo fmt

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,124 +1,424 @@
...
+2. **Add tests**: Include unit tests for new functionality
```
Comment (humbertoyusta) [inline CONTRIBUTING.md:314]: this does not make sense for new models/providers

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,124 +1,424 @@
...
+### Common Issues
 
+**Model not appearing in capabilities:**
+- Check if it's added to `running_models` in provider config
+- Verify the model exists in `known_models.json`
+- Ensure provider is properly loaded
+
+**Tokenizer errors:**
+- Verify tokenizer path is correct
+- Check if tokenizer supports the model's special tokens
```
Comment (humbertoyusta) [inline CONTRIBUTING.md:328]: this exact one is not needed, only keep very tokenizer path and use fake for testing

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,124 +1,424 @@
...
+- Use `fake` tokenizer for testing if needed
+
+**API connection issues:**
+- Verify endpoint URLs are correct
+- Check API key format and permissions
```
Comment (humbertoyusta) [inline CONTRIBUTING.md:333]: change the "and permissions" to "authorization"

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -1,124 +1,424 @@
...
+- Ensure FIM tokens are correctly configured
+- Check `scratchpad` type is appropriate
+- Verify context format matches model expectations
+
+### Debug Commands
 
 ```bash
-cargo build && target/debug/refact-lsp --http-port 8001 --reset-memory --experimental --workspace-folder . --logs-stderr --vecdb --ast
+# Check logs
+cargo run -- --logs-stderr --experimental
```
Comment (humbertoyusta) [inline CONTRIBUTING.md:345]: rm this part of Check logs

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -1,124 +1,424 @@
...
+
+# Validate configuration
+cargo check
+```
+
+## 💡 Examples
+
+### Example 1: Adding Claude 4 (Hypothetical) 
+
+1. **Add to known_models.json:**
```
Comment (humbertoyusta) [inline CONTRIBUTING.md:359]: rm this part, no need to add to known models.json

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -1,124 +1,424 @@
...
+  "supports_multimodality": true,
+  "supports_agent": true,
+  "scratchpad": "PASSTHROUGH",
+  "tokenizer": "hf://Xenova/claude-tokenizer"
+}
+```
+
+2. **Update anthropic.yaml:**
+```yaml
+running_models:
```
Comment (humbertoyusta) [inline CONTRIBUTING.md:373]: then in here, you need to add the n_ctx and all parts in the yaml

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -1,124 +1,424 @@
...
 ```
 
-Running those tests is still manual. To make sure your work didn't break other features,
-run tests for things you might have broken.
+---
+
+## 🎯 Next Steps
+
+1. **Join our [Discord](https://www.smallcloud.ai/discord)** for community support
+2. **Check [GitHub Issues](https://github.com/smallcloudai/refact/issues)**  *new-modals* tags for contribution opportunities
```
Comment (humbertoyusta) [inline CONTRIBUTING.md:418]: new-modals? is it a typo?

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -1,124 +1,365 @@
...
+-  **Update anthropic.yaml:**
+```yaml
+running_models:
+  - claude-4
+  - claude-3-7-sonnet-latest
+
+models:
+  claude-4:
+    n_ctx: 200000
+    generation_params:
```
Comment (humbertoyusta) [inline CONTRIBUTING.md:309]: these fields are incorrect, claude 4 should have fields like:

"n_ctx": 200000,
            "supports_tools": true,
            "supports_multimodality": true,
            "supports_agent": true,
            "supports_reasoning": "anthropic",
            "supports_boost_reasoning": true,
            "tokenizer": "hf://Xenova/claude-tokenizer"
            
            (in yaml, not json)

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -1,124 +1,365 @@
...
+      temperature: 0.7
+      top_p: 0.95
+  claude-3-7-sonnet-latest:
+    n_ctx: 100000
+    generation_params:
```
Comment (humbertoyusta) [inline CONTRIBUTING.md:314]: Same here

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -1,124 +1,365 @@
...
+
+```yaml
+# custom-ai.yaml
+chat_endpoint: https://api.custom-ai.com/v1/chat/completions
+supports_completion: false
+api_key: ca-...
+
+running_models:
+  - custom-model-pro
+  - custom-model-lite
```
Comment (humbertoyusta) [inline CONTRIBUTING.md:345]: here apart from these, the models should be configured, example, like claude-4

**Review by humbertoyusta — CHANGES_REQUESTED** [review]

**Review by humbertoyusta — APPROVED** [review]
